### ISSUE TITLE
batching_method=random dont seem to work properly 

### ISSUE THREAD

**Issue by oscarfossey** [mosaicml/streaming#657]
## Environment

- OS: [Ubuntu 20.04]
- Hardware (GPU, or instance type): [T4]
- mosaicml-streaming             0.7.4
- streaming                      0.1.2

## To reproduce

Thank you for the great repo.
I am trying to stream data from multiple sources i have the following code:
```
streams = [
        Stream(
            remote=remote_uris[i],
            local=os.path.join(local_dir, local_dirs_suffixes[i]),
            split=None,
            proportion=0.1 #for 10 streams
        ) for i in range(len(remote_uris))
    ]

dataset = StreamingDataset(
                streams=streams, #10 different streams
                batch_size=64,
                batching_method="random",
                predownload=8*64,
                shuffle=True,
                shuffle_algo="py1s",
                shuffle_block_size=100_000,
                num_canonical_nodes=1,
                cache_limit="100gb",
                epoch_size=10*64,
            )

dataloader = StreamingDataLoader(
               dataset = dataset,
               batch_size = 64,
               shuffle = False, 
               drop_last = True,
               num_workers = 8,
               prefetch_factor = 2
        )
```
When going through different batch from the dataloader, the batches dont seem correclty shuffled.

- batch1: [StreamC, StreamC, StreamC, StreamC, StreamC, StreamC, StreamD, ..., StreamC]
- batch2: [StreamF, StreamF, StreamF, StreamF, StreamF, StreamC, StreamD, ..., StreamF]
- batch3: [StreamB, StreamA, StreamB, StreamA, StreamB, StreamA, StreamB, ...., StreamA]
- ......

## Expected behavior

Real random distribution across the 10 different streams.
- batch1: [StreamC, StreamA, StreamB, StreamF, StreamE, StreamH, StreamD, ..., StreamC]
- batch2: [StreamF, StreamG, StreamD, StreamF, StreamA, StreamC, StreamB, ..., StreamB]
- batch3: [StreamE, StreamA, StreamB, StreamC, StreamD, StreamH, StreamB, ...., StreamE]
- ......

**Comment by snarayan21** [comment]
Hey @oscarfossey, seems like you’re setting num_canonical_nodes to 1 and using py1s shuffle. For the py1s and py2s shuffle algos, num_canonical_nodes is the only parameter that controls shuffle strength — shuffle_block_size is not used. So in this case, if you want to keep the py1s shuffling algo, increase num_canonical_nodes. Otherwise, switch to the py1e or py1br shuffle algos, and increase the shuffle_block_size to get a stronger shuffle.

Hope this helps!
